openmmapi/include/openmm/ConstantPotentialForce.h

```diff
@@ -0,0 +1,486 @@
+     * @param[out] particle1  the index of the first particle involved in the interaction
+     * @param[out] particle2  the index of the second particle involved in the interaction
+     * @param[out] chargeProd the scaled product of the atomic charges (i.e. the strength of the Coulomb interaction), measured in units of the proton charge squared
+     */
+    void getExceptionParameters(int index, int& particle1, int& particle2, double& chargeProd) const;
+    /**
+     * Set the force field parameters for an interaction that should be
+     * calculated differently from others.  If chargeProd is equal to 0, this
+     * will cause the interaction to be completely omitted from force and energy
+     * calculations.
+     * 
+     * Cutoffs are never applied to exceptions.  That is because they are
+     * primarily used for 1-4 interactions, which are really a type of bonded
+     * interaction and are parametrized together with the other bonded
+     * interactions.
+     *
+     * @param index      the index of the interaction for which to get parameters
+     * @param particle1  the index of the first particle involved in the interaction
+     * @param particle2  the index of the second particle involved in the interaction
+     * @param chargeProd the scaled product of the atomic charges (i.e. the strength of the Coulomb interaction), measured in units of the proton charge squared
+     */
+    void setExceptionParameters(int index, int particle1, int particle2, double chargeProd);
+    /**
+     * Identify exceptions based on the molecular topology.  Particles which are separated by one or two bonds are set
+     * to not interact at all, while pairs of particles separated by three bonds (known as "1-4 interactions") have
+     * their Coulomb interactions reduced by a fixed factor.
+     *
+     * @param bonds           the set of bonds based on which to construct exceptions.  Each element specifies the indices of
+     *                        two particles that are bonded to each other.
+     * @param coulomb14Scale  pairs of particles separated by three bonds will have the strength of their Coulomb interaction
+     *                        multiplied by this factor
+     */
+    void createExceptionsFromBonds(const std::vector<std::pair<int, int> >& bonds, double coulomb14Scale);
+    /**
+     * Update the particle and exception parameters in a Context to match those
+     * stored in this Force object.  This method provides an efficient method to
+     * update certain parameters in an existing Context without needing to
+     * reinitialize it.  Simply call setParticleParameters() and
+     * setExceptionParameters() to modify this object's parameters, then call
+     * updateParametersInContext() to copy them over to the Context.
+     *
+     * This method has several limitations.  The only information it updates is
+     * the parameters of particles and exceptions.  All other aspects of the
```

Comment:
Should it also allow changing electrode parameters and the external field?  I can imagine someone might want to vary those.